src/NBitcoin.Tests/ChainTests.cs

```diff
@@ -461,20 +454,123 @@ public ChainedBlock AppendBlock(ChainedBlock previous, params ConcurrentChain[]
         {
             ChainedBlock last = null;
             var nonce = RandomUtils.GetUInt32();
-            foreach(var chain in chains)
+            foreach (var chain in chains)
             {
                 var block = TestUtils.CreateFakeBlock(new Transaction());
                 block.Header.HashPrevBlock = previous == null ? chain.Tip.HashBlock : previous.HashBlock;
                 block.Header.Nonce = nonce;
-                if(!chain.TrySetTip(block.Header, out last))
+                if (!chain.TrySetTip(block.Header, out last))
                     throw new InvalidOperationException("Previous not existing");
             }
             return last;
         }
+
         private ChainedBlock AppendBlock(params ConcurrentChain[] chains)
         {
             ChainedBlock index = null;
             return AppendBlock(index, chains);
         }
+
+        /// <summary> 
+        /// Adapted from bitcoin core test, verify GetAncestor is using skip list in <see cref="ChainedBlock"/>.
+        /// <seealso cref="https://github.com/bitcoin/bitcoin/blob/master/src/test/skiplist_tests.cpp"/>
+        /// </summary>
+        [Fact]
+        [Trait("UnitTest", "UnitTest")]
+        public void ChainedBlockVerifySkipListForGetAncestor()
+        {
+            const int skipListLength = 300000;
```

Comment:
probably no need for const here, constants don't really belong here and should have modifiers,
i'd just use int variable ...

Reply:
Ok will do.